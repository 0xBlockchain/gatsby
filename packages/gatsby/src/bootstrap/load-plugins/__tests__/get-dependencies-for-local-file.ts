import getDependencies from "../get-dependencies-for-local-file"
import path from "path"

describe(`gets dependencies`, () => {
  it(`gets dependencies with file hash`, async () => {
    const result = await getDependencies(
      __dirname,
      path.join(`fixtures`, `file-with-node-dependency.ts`)
    )
    expect(result).toMatchSnapshot()

    const result2 = await getDependencies(
      __dirname,
      path.join(`fixtures`, `requires-other-local-file.js`)
    )
    expect(result2).toMatchSnapshot()
    expect(Object.keys(result2)).toHaveLength(2)
  })
  it(`gets npm dependencies with version`, async () => {
    const result = await getDependencies(
      __dirname,
      path.join(`fixtures`, `has-dependency.js`)
    )
    // Strange way perhaps to test this is a version string e.g. 8.1.2
    expect(result.precinct.split(`.`)).toHaveLength(3)
  })
})
