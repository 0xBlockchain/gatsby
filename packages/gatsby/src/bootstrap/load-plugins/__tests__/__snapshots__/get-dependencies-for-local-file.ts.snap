// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets dependencies gets dependencies with file hash 1`] = `
Object {
  "fixtures/file-with-node-dependency.ts": "6562831d242897e3c2caedf33beaf4d645945ecd",
}
`;

exports[`gets dependencies gets dependencies with file hash 2`] = `
Object {
  "fixtures/file-with-node-dependency.ts": "6562831d242897e3c2caedf33beaf4d645945ecd",
  "fixtures/requires-other-local-file.js": "7ae35bd4b775e72dd91c9a2af2441537575111ba",
}
`;
